### PR TITLE
DataRegistryUtils

### PR DESCRIPTION
**Pull request by claireh93**
Implement DataRegistry utils in `Scotland_run.jl` and remove SimulationData. Use sql to aggregate the Scottish population array by age category.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,6 +27,18 @@ deps/src/
 docs/build/
 docs/site/
 
+# Files generated when running data pipeline or saving output
+access*
```
Comment (richardreeve) [inline .gitignore]: That seems like a dangerous pattern for .gitignore

File: .gitignore [hunk 2]
```diff
@@ -27,6 +27,18 @@ deps/src/
...
+**/*.jld
+**/*.jlso
+**/*.pdf
+**/*.csv
+**/*.txt
```
Comment (richardreeve) [inline .gitignore]: These seem like dangerous patterns for .gitignore too

File: .gitignore [hunk 3]
```diff
@@ -27,6 +27,18 @@ deps/src/
...
+**/*.h5
+examples/**/*.toml
+out/**/*.toml
+**/*.db
+
```
Comment (richardreeve) [inline .gitignore:32]: Generally, I'm not sure what's going on here. Do we need to fill the repo with all of these files and then ignore them? The whole thing makes me a bit nervous...

File: Project.toml [hunk 4]
```diff
@@ -52,11 +52,8 @@ Unitful = "1986cc42-f94f-5a68-af5c-568840ba703d"
 
 [compat]
 ArchGDAL = "~0.4.0"
-DataAPI = "=1.2.0"
-HDF5 = "0.13.2"
+HDF5 = "^0.15.0"
```
Comment (richardreeve) [inline Project.toml:56]: Won't we need compat limits on everything? Should you add a CompatHelper github action that watches dev (a la Phylo.jl) to fix the rest?

File: examples/Epidemiology/data_config.yaml [hunk 5]
```diff
@@ -5,7 +5,17 @@ namespace: SCRC
 
 read:
   - where:
-      data_product: human/infection/SARS-CoV-2/*
```
Comment (richardreeve) [inline examples/Epidemiology/data_config.yaml:8]: Does * not work any longer?

File: test/simulationdata/test_integration.jl [hunk 6]
```diff
@@ -1,47 +1,47 @@
...
-            @test_broken isfile("simulationdata/demographics/human/demographics/population/scotland/1.0.0/1.0.0.h5")
-            @test_broken size(scotpop) == (465, 691, 10)
-            @test_broken sum(scotpop .< 0) == 0
-            @test_broken sum(scotpop) > 5e6
-        finally
-            remove_data()
-        end
-    end
-end
+# using SimulationData
```
Comment (richardreeve) [inline test/simulationdata/test_integration.jl]: I guess you just delete this file?

File: src/Epidemiology/EpiHelper.jl [hunk 7]
```diff
@@ -166,20 +167,19 @@ function initialise_output_abuns(
     # - initialise HDF5 file
     h5open(h5fn, "w") do fid
```
Comment (richardreeve) [inline src/Epidemiology/EpiHelper.jl:163]: I'm not sure what's going on here... don't we have write working yet in the pipeline? I thought that was now fixed?

**Review by richardreeve — CHANGES_REQUESTED** [review]
I obviously haven't looked at the new DataRegistry interface yet, but this looks more complex than I was hoping I have to admit. Anyway, something for when we have time to get together?

File: src/Epidemiology/EpiHelper.jl (hunk 7 above)
Comment (claireh93) [reply src/Epidemiology/EpiHelper.jl:163]: I don't think so yet, but at any rate, we can merge this in on a subsequent PR.

**Pull request closed without merging**